### PR TITLE
Cache trig tables and evaluate polylog in more cases

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed

Try caching and reusing dicts.

Evaluate a few more `polylog` cases.

#### Release Notes

* functions
   * `polylog` evaluates for a few more cases.

### CODE REVIEW

**Comment by oscargus** [comment]
Restarting to see if #23940 is really required or if the newly released setuptools solves it.

**Comment by oscargus** [comment]
Will self-merge in a week unless there are some comments.

File: sympy/functions/elementary/trigonometric.py [hunk 1]
```diff
@@ -108,6 +109,24 @@ def _period(self, general_period, symbol=None):
         raise NotImplementedError("Use the periodicity function instead.")
 
 
+@cacheit
+def _table2():
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/trigonometric.py:113]: Why not just make this a global variable like:
```
_TABLE2 = {12: (3, 4), ...}
```

Comment (oscarbenjamin) [reply sympy/functions/elementary/trigonometric.py:113]: Actually I suppose having it as a function means that there's less import time overhead and reduced memory if it never actually gets used.

**Comment by oscarbenjamin** [comment]
Looks good.

**Pull request merged**